ci/lava: allow resubmitting LAVA Test failures

mwasilew opened the pull request:
This patch broadens the list of potential test job error messages that
qualify for automated resubmission. Also messages can now be added to
LAVA ci backend setting as regular expressions.

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

terceiro commented on squad/ci/backend/lava.py:
```diff
@@ -327,8 +328,10 @@ def __parse_results__(self, data, test_job):
+                        infra_messages_re = re.compile("|".join(lava_infra_error_messages))
+                        if metadata['error_type'] in ['Infrastructure', 'Job', 'Test'] and \
+                                len(lava_infra_error_messages) > 0 and \
+                                infra_messages_re.search(metadata['error_msg']) is not None:
```
are you dropping the case insensitivity (`.lower()`) on purpose?

terceiro replied:
you should probably handle invalid regular expressions

terceiro replied:
I'm in doubt wrt joining all the items like with (`"|".join(...)`). if one of the items is not a valid regular expression, you will invalidate the entire check. I would keep testing one by one, and ignoring the ones that are not valid regexes.

mwasilew commented:
I added re.I so the regex won't be case sensitive. Also added a check for incorrect expressions which are ignored. 

I had issues saving \d (or similar) strings into YAML in the tests. Not sure if that works when added in the admin interface.
